test/tests/functional/pbs_hook_set_jobenv.py

```diff
@@ -180,32 +182,40 @@ def common_log_match(self, daemon):
                   "TEST_SQUOTE5=music \\'makes \\\"the\\'\\\"people",
                   "TEST_SQUOTE6=loving\\'",
                   "TEST_SPECIAL={}[]()~@#$%^&*!",
-                  "TEST_SPECIAL2=<dumb-test_text>"]
+                  "TEST_SPECIAL2=<dumb-test_text>",
+                  "TEST_RETURN=\\'3\,",
+                  "4\,",
+                  "5\\',"]
 
         if (daemon == "mom"):
             self.logger.info("Matching in mom logs")
+            logfile_type = self.mom
         elif (daemon == "server"):
             self.logger.info("Matching in server logs")
+            logfile_type = self.server
         else:
             self.logger.info("Provide a valid daemon name; server or mom")
             return
-
+        lines = None
+        search_msg = 'log match: searching for '
+        nomatch_msg = ' No match for '
         for msg in logmsg:
-            if (daemon == "mom"):
-                # Match the trailing separator (',')
-                self.mom.log_match(msg + ',')
-            elif (daemon == "server"):
-                self.server.log_match(msg, starttime=self.server.ctime)
-
-        # Following lines are commented due to PTL bug PP-1008
-        # if (daemon == "mom"):
-        #    rv = self.mom.log_match(
-        #          msg="TEST_RETURN\=\\\'3\\\,\n4\\\,\n5\\\'", regexp=True)
-        #    self.assertTrue(rv)
-        # else:
-        #    rv = self.server.log_match(
-        #            msg="TEST_RETURN\=\\\'3\\\,\n4\\\,\n5\\\'", regexp=True)
-        #    self.assertTrue(rv)
+            attempt = 1
+            while attempt <= 60:
+                lines = self.server.log_lines(
+                    logfile_type, starttime=self.server.ctime)
+                match = logutils.match_msg(lines, msg=msg)
+                if match:
+                    self.logger.info(search_msg + msg + ' ... OK')
+                    break
+                else:
+                    self.logger.info(nomatch_msg + msg +
+                                     ' attempt ' + str(attempt))
+                attempt += 1
+                time.sleep(0.5)
+            if (match is None):
```

Comment:
You don't need parentheses around this, just `if match is None:`

Reply:
Done